src/impl-genera.lisp

```diff
@@ -92,11 +92,11 @@ Distributed under the MIT license (see LICENSE file)
   (check-type lock recursive-lock)
   (check-type timeout (or null (satisfies zerop)) "Unsupported :TIMEOUT value")
   (when (or (null timeout)
-	    (process:lock-lockable-p (recursive-lock-lock lock)))
+            (process:lock-lockable-p (recursive-lock-lock lock)))
```

Comment:
Same as for `acquire-lock`, there's a race condition here as well.
Can you also refactor `acquire-lock` into `acquire-lock-internal`, which takes an additional `wait-p` argument ? Same for `acquire-recursive-lock`.
That way the call to `process:lock-lockable-p` would go in the internal function, and `acquire-lock` and`acquire-recursive-lock` would be much more similar.

Reply:
I have made the change requested here.  Rather than give ACQUIRE-RECURSIVE-LOCK-INTERNAL a WAIT-P argument, I gave it a TIMEOUT argument in the event that I can someday figure out a way to implement it.